app/api/v1/routes/auth.py

```diff
@@ -24,56 +26,72 @@
 async def login(
     db=Depends(get_db), form_data: OAuth2PasswordRequestForm = Depends()
 ):
-    user = authenticate_user(db, form_data.username, form_data.password)
-    if not user:
-        raise HTTPException(
-            status_code=status.HTTP_401_UNAUTHORIZED,
-            detail="Incorrect username or password",
-            headers={"WWW-Authenticate": "Bearer"},
+    try:
+        user = authenticate_user(db, form_data.username, form_data.password)
+        if not user:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Incorrect username or password",
+                headers={"WWW-Authenticate": "Bearer"},
+            )
+
+        access_token_expires = timedelta(
+            minutes=security.ACCESS_TOKEN_EXPIRE_MINUTES
+        )
+        if user.is_superuser:
+            permissions = "admin"
+        else:
+            permissions = "user"
+        access_token = security.create_access_token(
+            data={"sub": user.email, "permissions": permissions},
+            expires_delta=access_token_expires,
         )
 
-    access_token_expires = timedelta(
-        minutes=security.ACCESS_TOKEN_EXPIRE_MINUTES
-    )
-    if user.is_superuser:
-        permissions = "admin"
-    else:
-        permissions = "user"
-    access_token = security.create_access_token(
-        data={"sub": user.email, "permissions": permissions},
-        expires_delta=access_token_expires,
-    )
+        return {"access_token": access_token, "token_type": "bearer", "permissions": permissions}
 
-    return {"access_token": access_token, "token_type": "bearer", "permissions": permissions}
+    except Exception as e:
```

Comment:
Fast API docs expects the HTTP exception to be raised in case login error.
```
except HTTPException as e:
   # handle Exception from line 32
    raise e
except Exception as e:
   # catch all
   ...
```

Reply:
I would really like to get back to using raise...
I will make another PR addressing raise vs. return, but you're 100% correct that we should be using raise like this